cpp/velox/operators/writer/VeloxParquetDatasource.cc

```diff
@@ -49,7 +50,10 @@ void VeloxParquetDatasource::init(const std::unordered_map<std::string, std::str
     sink_ = std::make_unique<WriteFileSink>(std::move(localWriteFile), path);
   } else if (strncmp(filePath_.c_str(), "s3a:", 4) == 0) {
 #ifdef ENABLE_S3
-    auto fileSystem = getFileSystem(filePath_, nullptr);
+    auto confs = std::make_shared<facebook::velox::core::MemConfigMutable>(sparkConfs);
+    auto hiveConfs = getHiveConfig(confs);
+    auto fileSystem =
+        getFileSystem(filePath_, std::make_shared<facebook::velox::core::MemConfig>(hiveConfs->valuesCopy()));
     auto* s3FileSystem = dynamic_cast<filesystems::S3FileSystem*>(fileSystem.get());
     sink_ = std::make_unique<dwio::common::WriteFileSink>(
         s3FileSystem->openFileForWrite(filePath_, {{}, s3SinkPool_.get()}), filePath_);
```

Comment:
@dcoliversun Thanks for refine the conf extractor. Here is a legacy issue that the `WriteFileSink` is not the API Velox supposed to expose as there is also a [FileSinkFactory] (https://github.com/facebookincubator/velox/blob/5e0fd098e285beafea5890dd1c0991b0dc5f3378/velox/connectors/hive/storage_adapters/s3fs/RegisterS3FileSystem.cpp#L85) registered, can you help try following API?
```
    auto confs = std::make_shared<facebook::velox::core::MemConfigMutable>(sparkConfs);
    auto hiveConfs = getHiveConfig(confs);
    sink_ = dwio::common::FileSink::create(filePath_, {.connectorProperties = hiveConfs, .pool = s3SinkPool_.get()});
```

Reply:
Thanks! It works as expected and code updated

Reply:
BTW, should all sinks use `FileSink` ?

Reply:
Yes. I just confirmed that CI/CD can test `LocalFileSink`/`HDFSFileSink` path, can you help update them too?

Reply:
Sure, I'd like to do this. Is it better to open other pr to do this? Because this pr only aims to support s3 filesystem.

Reply:
Sure.